src/pitaya/src/ffi.rs

```diff
@@ -826,14 +835,14 @@ pub extern "C" fn pitaya_metrics_observe_hist(
     name: *mut c_char,
     value: f64,
     labels: *mut *mut c_char,
-    labels_count: usize,
+    labels_count: u32,
     callback: extern "C" fn(*mut c_void),
     user_data: *mut c_void,
 ) {
     let p = unsafe { mem::ManuallyDrop::new(Box::from_raw(p)) };
     let name = unsafe { c_string_to_string(name) };
     let labels: Vec<String> = unsafe {
-        slice::from_raw_parts(labels, labels_count)
+        slice::from_raw_parts(labels, labels_count as usize)
```

Comment:
Shouldn't you use an `u32` type?

Reply:
No, this function expects a `usize` instead of a `u32`. The conversion is also fine, since `u32` is smaller or equal than a `usize`.